http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/RuntimeServiceInterp.scala

```diff
@@ -0,0 +1,316 @@
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+import org.broadinstitute.dsde.workbench.model.{google, TraceId, UserInfo, WorkbenchEmail}
+
+import scala.concurrent.ExecutionContext
+
+class RuntimeServiceInterp[F[_]](blocker: Blocker,
+                                 semaphore: Semaphore[F],
+                                 config: RuntimeServiceConfig,
+                                 authProvider: LeoAuthProvider[F],
+                                 serviceAccountProvider: ServiceAccountProvider[F],
+                                 dockerDAO: DockerDAO[F],
+                                 googleStorageService: GoogleStorageService[F],
+                                 publisherQueue: fs2.concurrent.Queue[F, LeoPubsubMessage])(implicit F: Async[F],
+                                                                                            log: StructuredLogger[F],
+                                                                                            timer: Timer[F],
+                                                                                            cs: ContextShift[F],
+                                                                                            dbReference: DbReference[F],
+                                                                                            ec: ExecutionContext)
+    extends RuntimeService[F] {
+
+  def createRuntime(userInfo: UserInfo,
+                    googleProject: GoogleProject,
+                    runtimeName: RuntimeName, //TODO: rename this to RuntimeName once Rob's PR is in
+                    req: CreateRuntime2Request)(implicit as: ApplicativeAsk[F, RuntimeServiceContext]): F[Unit] =
+    for {
+      context <- as.ask
+      implicit0(traceId: ApplicativeAsk[F, TraceId]) <- F.pure(ApplicativeAsk.const[F, TraceId](context.traceId)) //TODO: is there a better way to do this?
+      hasPermission <- authProvider.hasProjectPermission(userInfo, ProjectActions.CreateClusters, googleProject)
+      _ <- if (hasPermission) F.unit else F.raiseError[Unit](AuthorizationError(Some(userInfo.userEmail)))
+      // Grab the service accounts from serviceAccountProvider for use later
+      clusterServiceAccountOpt <- serviceAccountProvider
+        .getClusterServiceAccount(userInfo, googleProject)
+      notebookServiceAccountOpt <- serviceAccountProvider
+        .getNotebookServiceAccount(userInfo, googleProject)
+      serviceAccountInfo = ServiceAccountInfo(clusterServiceAccountOpt, notebookServiceAccountOpt)
+
+      clusterOpt <- clusterQuery.getActiveClusterByNameMinimal(googleProject, runtimeName).transaction
+
+      _ <- clusterOpt match {
+        case Some(c) => F.raiseError[Unit](RuntimeAlreadyExistsException(googleProject, runtimeName, c.status))
+        case None =>
+          for {
+            internalId <- F.pure(RuntimeInternalId(context.traceId.asString)) //TODO: is there any downside to reusing traceId as internalId?
```

Comment:
Hm.. this is also used as a Sam id, are there security issues with exposing internal Sam resource ids in error messages, etc?

Reply:
I'll use UUID then just in case